WEB: Restructuring pages and navigation, styling and new footer

datapythonista opened the pull request:
xref #28519
closes #28520

You can see the result rendered in: https://datapythonista.github.io/pandas-web/ (the new logo is not in the PR).

Been discussing with NumPy the navigation, and what is proposed here is with minimal differences agreed with them. See https://github.com/numpy/numpy.org/issues/43

I don't touch the home page here, I think we need to remove almost everything we've got so far (based on the current), and move more in the direction of Dask or Jupyter. Discussions about it better in #28168 than here.

datapythonista committed:
More restructuring and styling

TomAugspurger commented on web/pandas/_templates/layout.html:
```diff
@@ -20,41 +20,44 @@
             <link rel="stylesheet"
                   href="{{ base_url }}{{ stylesheet }}">
         {% endfor %}
+        <script src="https://kit.fontawesome.com/79e5369384.js" crossorigin="anonymous"></script>
```
My understanding is that loading stylesheets and JS should go at the bottom of the page, to not slow down the loading of the page content. Is that accurate?

datapythonista replied:
I followed fontawesome instructions, I don't know if there is a reason to have it in the header, or it doesn't matter, but it's where they ask to add it.

Not an expert, but I'd say browsers load those asynchronously. I assumed that they were placed at the end to have the dom loaded when the code is executed. But my knowledge about web is outdated 10 years, so don't really know.

Thanks for catching the broken link, just fixed them.

TomAugspurger reviewed (commented):
Just FYI: the links pointing to, e.g., "install" will need to be updated.

WillAyd commented:
Looks nice - great work @datapythonista

datapythonista committed:
Updating links to install

jorisvandenbossche commented:
Looks nice indeed!

I see that you changed the "Install" page to a "Getting Started" page, with some additional links to external resources. So that is something we we will need to coordinate with the work @stijnvanhoey is doing on the getting started landing page in https://github.com/pandas-dev/pandas/issues/26831. 

Will try to take a closer look tomorrow.

datapythonista commented:
Didn't know that @stijnvanhoey was working on that already, but in the Tutorials section (https://datapythonista.github.io/pandas-web/getting_started.html#tutorials) I just have that sentence as a placeholder so he could have the list of pages he's working on, any anything else.

So I guess merging should be as easy as replacing the content of that section for whatever he's got.

jorisvandenbossche commented:
It's part of the grant proposal, and just also started on that this week (I think Stijn will open a PR once he has something to show). The idea is very similar as you have here: short install instructions, overview tutorials, links to extra resources. So indeed, I don't think there is any conflict in the work being done (unless you were planning to add more than what there is now in the PR?). 

Except for the question (again) where to put it: in the sphinx docs or markdown website. I am also fine with having it on the main website (anyway, if it is in the docs, then there should also need to be a clear link to that page). But, we also need some landing page for the getting started section in the docs as well (to not have the dry table of contents as there is now (https://pandas.pydata.org/pandas-docs/stable/getting_started/index.html), so need to think a bit about what to put where (or what to duplicate).

datapythonista commented:
Makes sense. And I'm not planning to add anything else to the getting
started.

I think it makes more sense to have it in the website, since we don't want
to wait for a release to change this page, and the website will be updated
immediately. If there is a way to update parts of the docs on commits to
master without making things trickier to manage, not a problem to have it
in the docs either. But I also prefer markdown over rst.

Any objection on merging this? I think there is agreement on the main idea,
and we'll have to keep making small changes to the website anyway.

On Wed, Sep 25, 2019 at 8:27 AM Joris Van den Bossche <
notifications@github.com> wrote:

> It's part of the grant proposal, and just also started on that this week
> (I think Stijn will open a PR once he has something to show). The idea is
> very similar as you have here: short install instructions, overview
> tutorials, links to extra resources. So indeed, I don't think there is any
> conflict in the work being done (unless you were planning to add more than
> what there is now in the PR?).
>
> Except for the question (again) where to put it: in the sphinx docs or
> markdown website. I am also fine with having it on the main website
> (anyway, if it is in the docs, then there should also need to be a clear
> link to that page). But, we also need some landing page for the getting
> started section in the docs as well (to not have the dry table of contents
> as there is now (
> https://pandas.pydata.org/pandas-docs/stable/getting_started/index.html),
> so need to think a bit about what to put where (or what to duplicate).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/28582?email_source=notifications&email_token=ACMXUAEUGWFB4GNLR6X44L3QLNDI3A5CNFSM4IZOJRJKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD7RRVEQ#issuecomment-534977170>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACMXUADVYFCGYFUGR3S7OX3QLNDI3ANCNFSM4IZOJRJA>
> .
>

jorisvandenbossche commented:
> Any objection on merging this?

I am doing a quick review as we speak

jorisvandenbossche commented on web/pandas/about/sponsors.md:
```diff
@@ -0,0 +1,33 @@
+# Sponsors
+
```
There is currently no mention of Tidelift? (here or on the donate or contribute pages)

(the current website has a very brief one: "You can support pandas by becoming a Tidelift subscriber." on the donate page

jorisvandenbossche commented on web/pandas/config.yml:
```diff
@@ -4,59 +4,60 @@ main:
     - name: "Release notes"
       target: /docs/whatsnew/index.html
+    - name: "Older versions"
+      target: https://pandas.pydata.org/pandas-docs/version/
```
this link does not work

jorisvandenbossche commented on web/pandas/config.yml:
```diff
@@ -4,59 +4,60 @@ main:
-      target: /community/team.html
-    - name: "Code of Conduct"
+    - name: "Discuss"
+      target: https://pandas.discourse.group
```
Maybe wait until some decision is taken? (since this website is not yet officially live, also fine to keep)

jorisvandenbossche commented on web/pandas/contribute.md:
```diff
@@ -0,0 +1,12 @@
+# Contribute to pandas
+
+_pandas_ is and always will be **free**. To make de development sustainable, we need _pandas_ users, corporate
```
```suggestion
_pandas_ is and always will be **free**. To make the development sustainable, we need _pandas_ users, corporate
```

jorisvandenbossche commented on web/pandas/config.yml:
```diff
@@ -4,59 +4,60 @@ main:
-  # - name: "Blog"
-  #  target: /blog.html
-  - name: "Donate"
-    target: /donate.html
```
Do we want to keep some direct link in the menu to this page? (now it is only linked to from other pages)

jorisvandenbossche commented on web/pandas/contribute.md:
```diff
@@ -0,0 +1,12 @@
+and about current sponsors in the [sponsors page](about/sponsors.html).
+Financial contributions will mainly be used to advance in the [pandas roadmap](about/roadmap.html).
+
+- If your **company or organization** is interested in helping make pandas better, please contact us at [info@numfocus.org](mailto:info@numfocus.org)
```
Just wondering: is this the best way to contact us? (we don't have that somewhere written now I think? But for sure we also don't have our "own" email address for pandas)

jorisvandenbossche commented on web/pandas/config.yml:
```diff
@@ -4,59 +4,60 @@ main:
+  - name: "About us"
+    target:
+    - name: "About pandas"
+      target: /about/about.html
```
Small nit: we could rename about/about.md to about/index.md? (the url would be a bit nicer I think)

jorisvandenbossche reviewed (commented):
Some small questions / comments

datapythonista commented:
Thanks for the review, good feedback.

I already have almost ready a new brach with the contribute page
refactored, where I delete the mail to numfocus and I link to the sponsors
page instead (I think that's where makes more sense to explain ways to
sponsor).

The donate link is more visible in the new version. I initially had it in
the navigation, but moved it there based on the preference from NumPy and
scikit-learn. I'm personally happy about that, it'll still be very easy to
find, and not create the wrong impression that we depend on the small
individual contributions.

I'll leave the contributing as is if that's ok, and we can fine tune the
next version which I think it's much better.

I don't have a page to link to the versions of the docs yet. Can replace
that broken link by a hash, but I thought that could be the right location,
and the link has to be broken for now, until we implement that page. I can
implement it in the web (and to make makes sense since I've got all the
versions from the github API already), but I think you wanted to have it in
the docs, right?

I'll address the rest of the comments, thanks!

On Wed, 25 Sep 2019, 12:58 Joris Van den Bossche, <notifications@github.com>
wrote:

> *@jorisvandenbossche* commented on this pull request.
>
> Some small questions / comments
> ------------------------------
>
> In web/pandas/about/sponsors.md
> <https://github.com/pandas-dev/pandas/pull/28582#discussion_r328074667>:
>
> > @@ -0,0 +1,33 @@
>
> +# Sponsors
>
> +
>
>
> There is currently no mention of Tidelift? (here or on the donate or
> contribute pages)
>
> (the current website has a very brief one: "You can support pandas by
> becoming a Tidelift subscriber." on the donate page
> ------------------------------
>
> In web/pandas/config.yml
> <https://github.com/pandas-dev/pandas/pull/28582#discussion_r328075599>:
>
> >      - name: "Release notes"
>
>        target: /docs/whatsnew/index.html
>
> +    - name: "Older versions"
>
> +      target: https://pandas.pydata.org/pandas-docs/version/
>
> this link does not work
> ------------------------------
>
> In web/pandas/config.yml
> <https://github.com/pandas-dev/pandas/pull/28582#discussion_r328075974>:
>
> >      - name: "Ask a question (StackOverflow)"
>
>        target: https://stackoverflow.com/questions/tagged/pandas
>
> -    - name: "Discuss (mailing list)"
>
> -      target: https://groups.google.com/forum/#!forum/pydata
>
> -    - name: "Team"
>
> -      target: /community/team.html
>
> -    - name: "Code of Conduct"
>
> +    - name: "Discuss"
>
> +      target: https://pandas.discourse.group
>
> Maybe wait until some decision is taken? (since this website is not yet
> officially live, also fine to keep)
> ------------------------------
>
> In web/pandas/contribute.md
> <https://github.com/pandas-dev/pandas/pull/28582#discussion_r328076252>:
>
> > @@ -0,0 +1,12 @@
>
> +# Contribute to pandas
>
> +
>
> +_pandas_ is and always will be **free**. To make de development sustainable, we need _pandas_ users, corporate
>
>
> ⬇️ Suggested change
>
> -_pandas_ is and always will be **free**. To make de development sustainable, we need _pandas_ users, corporate
>
> +_pandas_ is and always will be **free**. To make the development sustainable, we need _pandas_ users, corporate
>
>
> ------------------------------
>
> In web/pandas/config.yml
> <https://github.com/pandas-dev/pandas/pull/28582#discussion_r328076804>:
>
> >        target: /community/coc.html
>
> -    - name: "Citing pandas"
>
> -      target: /community/citing.html
>
> -  # - name: "Blog"
>
> -  #  target: /blog.html
>
> -  - name: "Donate"
>
> -    target: /donate.html
>
>
> Do we want to keep some direct link in the menu to this page? (now it is
> only linked to from other pages)
> ------------------------------
>
> In web/pandas/contribute.md
> <https://github.com/pandas-dev/pandas/pull/28582#discussion_r328077287>:
>
> > @@ -0,0 +1,12 @@
>
> +# Contribute to pandas
>
> +
>
> +_pandas_ is and always will be **free**. To make de development sustainable, we need _pandas_ users, corporate
>
> +or individual, to support the development by providing their time and money.
>
> +
>
> +You can find more information about current developers in the [team page](about/team.html),
>
> +and about current sponsors in the [sponsors page](about/sponsors.html).
>
> +Financial contributions will mainly be used to advance in the [pandas roadmap](about/roadmap.html).
>
> +
>
> +- If your **company or organization** is interested in helping make pandas better, please contact us at [info@numfocus.org](mailto:info@numfocus.org)
>
>
> Just wondering: is this the best way to contact us? (we don't have that
> somewhere written now I think? But for sure we also don't have our "own"
> email address for pandas)
> ------------------------------
>
> In web/pandas/config.yml
> <https://github.com/pandas-dev/pandas/pull/28582#discussion_r328079003>:
>
> >    css:
>
>    - /static/css/pandas.css
>
>  navbar:
>
> -  - name: "Install"
>
> -    target: /install.html
>
> +  - name: "About us"
>
> +    target:
>
> +    - name: "About pandas"
>
> +      target: /about/about.html
>
>
> Small nit: we could rename about/about.md to about/index.md? (the url
> would be a bit nicer I think)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/28582?email_source=notifications&email_token=ACMXUACHEB3Z2DAVXZCSSZ3QLNG4XA5CNFSM4IZOJRJKYY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOCF3QXAA#pullrequestreview-293014400>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACMXUAGZSCRUQE4NONYN3I3QLNG4XANCNFSM4IZOJRJA>
> .
>

jorisvandenbossche commented:
> I'll leave the contributing as is if that's ok, and we can fine tune the
next version which I think it's much better.


Sure!

> I don't have a page to link to the versions of the docs yet. Can replace
that broken link by a hash, but I thought that could be the right location,
and the link has to be broken for now, until we implement that page. I can
implement it in the web (and to make makes sense since I've got all the
versions from the github API already), but I think you wanted to have it in
the docs, right?

It's fine to leave the link there right now, it's a good reminder to do something abut this ;) 
I would like to have a dropdown or some widget in the docs to access other versions (similarly to how its on python or readthedocs). But that doesn't necessarily need to conflict with also a page with an overview of the doc pages (like https://scikit-learn.org/dev/versions.html)

datapythonista commented:
Cool, I'll implement that page then. It's fast, got everything I need in
the home I'm replacing. Will add it in the follow up PR.

On Wed, 25 Sep 2019, 13:40 Joris Van den Bossche, <notifications@github.com>
wrote:

> I'll leave the contributing as is if that's ok, and we can fine tune the
> next version which I think it's much better.
>
> Sure!
>
> I don't have a page to link to the versions of the docs yet. Can replace
> that broken link by a hash, but I thought that could be the right location,
> and the link has to be broken for now, until we implement that page. I can
> implement it in the web (and to make makes sense since I've got all the
> versions from the github API already), but I think you wanted to have it in
> the docs, right?
>
> It's fine to leave the link there right now, it's a good reminder to do
> something abut this ;)
> I would like to have a dropdown or some widget in the docs to access other
> versions (similarly to how its on python or readthedocs). But that doesn't
> necessarily need to conflict with also a page with an overview of the doc
> pages (like https://scikit-learn.org/dev/versions.html)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/28582?email_source=notifications&email_token=ACMXUAFDWXMDR54YI6JKJNLQLNL3HA5CNFSM4IZOJRJKYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD7RXZMY#issuecomment-535002291>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ACMXUACXH2LWANYZANYMGA3QLNL3HANCNFSM4IZOJRJA>
> .
>

datapythonista committed:
Fix typo in contribute.md

Co-Authored-By: Joris Van den Bossche <jorisvandenbossche@gmail.com>

datapythonista committed:
PR comments, adding Tidelift and renaming about/about to about/index

datapythonista commented:
@jorisvandenbossche addressed the comments. For the contribute page, as I said, have some work in a follow up branch that will remove that email, so I prefer to not touch it here to avoid conflicts.

I'll add to that PR the page with the old docs links too.

jorisvandenbossche commented:
Thanks!

Can you answer on:

> [about info@numfocus.org] Just wondering: is this the best way to contact us? (we don't have that somewhere written now I think? But for sure we also don't have our "own" email address for pandas)

datapythonista commented:
> Can you answer on:

Sorry, I thought I answered. For now I'm moving the problem from the contribute page to the sponsors page, see:

![contribute_screenshot](https://user-images.githubusercontent.com/10058240/65690998-80845700-e046-11e9-8295-d8ecdb6c3f3b.png)

That's why I didn't want to address this here, since touching that was just going to generate a conflict for nothing.

Then, I think we need something similar in the sponsors page (and probably worth adding a bit more information on what sponsors get, to manage expectations). On the form of contact, I'm fine with pointing directly to NumFOCUS, or may be we can have an address sponsoring@pandas.io that anyone interested in our team receives, and then we delegate to NumFOCUS? Or may be having whoever from our team and NumFOCUS is interested receiving that Inbox?

If it's technically possible to have a Discourse group receiving that address, and we can add to that group the people who should receive it, I think that would be the more efficient way of managing (assuming we move to Discourse). I'd also do the same then for coc@pandas.io, diversity@pandas.io, and whatever we may want.
